tests/unit/s2n_cipher_suite_match_test.c

```diff
@@ -76,6 +78,89 @@ int main(int argc, char **argv)
 
         EXPECT_EQUAL(count, S2N_CIPHER_SUITE_COUNT);
 
+        /* Test that the client allows the server to select ciphers that were offered in ClientHello */
+        {
+            EXPECT_SUCCESS(s2n_enable_tls13());
+            conn->client_protocol_version = S2N_TLS13;
+            conn->actual_protocol_version = S2N_TLS13;
+            conn->server_protocol_version = S2N_TLS13;
```

Comment:
It is NOT just copying tests. The tests were erroneous previously and had to be moved from its previous location, as `s2n_set_cipher_as_client` was being called on `S2N_SERVER` conn instead of `S2N_CLIENT`.